### ISSUE TITLE
Feature Request: TCN Layer (temporal convolutional networks)

### ISSUE THREAD

**Issue by shun-lin** [tensorflow/addons#661]
**Describe the feature and the current behavior/state.**

The feature is for adding the different layers introduced in [An Empirical Evaluation of Generic Convolutional and Recurrent Networks for Sequence Modeling](https://arxiv.org/pdf/1803.01271.pdf) paper to tfa.layers. There is already existed an Keras implementation here: https://github.com/philipperemy/keras-tcn/, I feel like this is helpful and would have more visibility in tfa so I ask him in this [issue](https://github.com/philipperemy/keras-tcn/issues/73) if I can help out porting the TCN layers portion into tfa and he says yes :)
If you feel like this would be a good addition let me know, very happy to contribute.

**Relevant information**
- Are you willing to contribute it (yes/no): yes of course :)
- Are you willing to maintain it going forward? (yes/no): yes of course :)
- Is there a relevant academic paper? (if so, where): yes [An Empirical Evaluation of Generic Convolutional and Recurrent Networks for Sequence Modeling](https://arxiv.org/pdf/1803.01271.pdf)
- Is there already an implementation in another framework? (if so, where):
No but was implemented in pytorch in this repo, https://github.com/locuslab/TCN, and Keras in this repo https://github.com/philipperemy/keras-tcn (will be working on porting the layers portion into tfa, asked owner for permission)
- Was it part of tf.contrib? (if so, where): no

**Which API type would this fall under (layer, metric, optimizer, etc.)**
layer

**Who will benefit with this feature?**
Developers who want to experiment with TCN layers.

**Any other info.**
Github issue that initialize the conversation to port TCN layers from @philipperemy's repo to tensorflow addons (see last two comments in this issue).
https://github.com/philipperemy/keras-tcn/issues/73

**Comment by WindQAQ** [comment]
Good news! This is indeed in the scope of addons. Feel free to file a PR for it. Thanks!

**Comment by shun-lin** [comment]
@WindQAQ thanks! Will do so when I have time :)

**Comment by seanpmorgan** [comment]
TensorFlow Addons is transitioning to a minimal maintenance and release mode. New features will not be added to this repository. For more information, please see our public messaging on this decision:
[TensorFlow Addons Wind Down](https://github.com/tensorflow/addons/issues/2807)

Please consider sending feature requests / contributions to other repositories in the TF community with a similar charters to TFA:
[Keras](https://github.com/keras-team/keras)
[Keras-CV](https://github.com/keras-team/keras-cv)
[Keras-NLP](https://github.com/keras-team/keras-nlp)
